src/GafferScene/Unencapsulate.cpp

```diff
@@ -0,0 +1,287 @@
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	return capsule->scene()->attributes( concatScenePath( capsule->root(), branchPath ) );
+}
+
+bool Unencapsulate::affectsBranchObject( const Gaffer::Plug *input ) const
+{
+	return input == inPlug()->objectPlug();
+}
+
+bool Unencapsulate::processesRootObject() const
+{
+    return true;
+}
+
+void Unencapsulate::hashBranchObject( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	IECore::ConstObjectPtr o = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( o.get() );
+ 	if( branchPath.size() == 0 )
+	{
+		if( capsule )
+		{
+			h = IECore::NullObject::defaultNullObject()->Object::hash();
+		}
+		else
+		{
+			h = inPlug()->objectPlug()->hash();
+		}
+	}
+	else
+	{
+		h = capsule->scene()->objectHash( concatScenePath( capsule->root(), branchPath ) );
+	}
+}
+
+IECore::ConstObjectPtr Unencapsulate::computeBranchObject( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context ) const
+{
+	IECore::ConstObjectPtr o = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( o.get() );
+ 	if( branchPath.size() == 0 )
+	{
+		// We're at a root location, not inside a capsule.  If there's a capsule here, we return a null
+		// object ( the capsule has now been expanded, so we don't want it in the scene any more ).
+		// If there's no capsule, we're not expanding anything, just pass through the source scene
+		if( capsule )
+		{
+			return IECore::NullObject::defaultNullObject();
+		}
+		else
+		{
+			return o;
+		}
+	}
+	return capsule->scene()->object( concatScenePath( capsule->root(), branchPath ) );
+}
+
+bool Unencapsulate::affectsBranchChildNames( const Gaffer::Plug *input ) const
+{
+	return input == inPlug()->objectPlug();
+}
+
+void Unencapsulate::hashBranchChildNames( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	if( !capsule )
+	{
+		h.append( 0 );
+		return;
+	}
+	if( branchPath.size() == 0 )
+	{
+		h = capsule->scene()->childNamesHash( capsule->root() );
+	}
+	else
+	{
+		h = capsule->scene()->childNamesHash( concatScenePath( capsule->root(), branchPath ) );
+	}
+}
+
+IECore::ConstInternedStringVectorDataPtr Unencapsulate::computeBranchChildNames( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context ) const
+{
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	if( !capsule )
+	{
+		return new InternedStringVectorData();
```

Comment:
Could use `outPlug()->childNamesPlug()->defaultValue()` here to avoid an allocation and match the suggestion in `hashBranchChildNames()`.